sensor_test.py

```diff
@@ -0,0 +1,178 @@
+):
+    """
+    1. Set sensor reading interval to 1.
+    2. Get sensor info.
+    3. Validate that sensor reading interval is set to interval from Step 1.
+    4. Get sensor reading.
+    5. Wait for interval specified in Step 1.
+    6. Get sensor reading.
+    7. Validate that reading from Step 4 doesn't equal reading from Step 6.
+    """
+
+    set_sensor_reading_interval(1)
+
+    assert (
+        get_sensor_info().get("reading_interval") == 1
+    ), "sensor reading interval is not set to interval from Step 1"
+
+    sensor_reading_before_wait = get_sensor_reading()
+    sleep(1)
+    sensor_reading_after_wait = get_sensor_reading()
+
+    assert (
+        sensor_reading_before_wait != sensor_reading_after_wait
+    ), "reading interval is same"
+
+    print("test_set_sensor_reading_interval passed")
+
+
+def test_update_sensor_firmware(get_sensor_info, update_sensor_firmware):
+    """
+    1. Get original sensor firmware version.
+    2. Request firmware update.
+    3. Get current sensor firmware version.
+    4. Validate that current firmware version is +1 to original firmware version.
+    5. Repeat steps 1-4 until sensor is at max_firmware_version - 1.
+    6. Update sensor to max firmware version.
+    7. Validate that sensor is at max firmware version.
+    8. Request another firmware update.
+    9. Validate that sensor doesn't update and responds appropriately.
+    10. Validate that sensor firmware version doesn't change if it's at maximum value.
+    """
+
+    max_firmware_version = 15
+
+    def update_firmware():
+        original_sensor_firmware_version = get_sensor_info().get("firmware_version")
+        update_sensor_firmware()
+
+        wait(
+            func=get_sensor_info,
+            condition=lambda x: isinstance(x, dict),
+            tries=13,
+            timeout=1,
+        )
+        current_sensor_firmware_version = get_sensor_info().get("firmware_version")
+
+        assert (
+            current_sensor_firmware_version == original_sensor_firmware_version + 1
+        ), "Firmware update is not successful"
+
+        return current_sensor_firmware_version
+
+    current_sensor_firmware_version = update_firmware()
+
+    while current_sensor_firmware_version < max_firmware_version - 1:
+        current_sensor_firmware_version = update_firmware()
+
+    current_sensor_firmware_version = update_firmware()
+
+    assert (
+        current_sensor_firmware_version == max_firmware_version
+    ), "Sensor is not at max firmware version"
+
+    another_firmware_update = update_sensor_firmware()
+    lasted_firmware_version_update = get_sensor_info().get("firmware_version")
+
+    assert (
+        another_firmware_update == "already at latest firmware version"
+    ), "Sensor does not respond properly"
+    assert (
+        current_sensor_firmware_version == lasted_firmware_version_update
+    ), "Error when updating the maximum version of the sensor firmware"
```

Comment:
This assert message needs clarification too. If this assert fails, we need to inform the user that sensor changed its version when it was at the max version already and received an update request. Again, longer messages are alright in cases like this.